components/lvgl_esp32_drivers/lvgl_touch/xpt2046.c

```diff
@@ -48,10 +48,16 @@ uint8_t avg_last;
  */
 void xpt2046_init(void)
 {
-    gpio_set_direction(XPT2046_IRQ, GPIO_MODE_INPUT);
-    gpio_set_direction(TP_SPI_CS, GPIO_MODE_OUTPUT);
-    gpio_set_level(TP_SPI_CS, 1);
-
+    gpio_config_t irq_config = {
+        .pin_bit_mask = 1UL << XPT2046_IRQ,
+        .mode = GPIO_MODE_INPUT,
+        .pull_up_en = GPIO_PULLUP_DISABLE,
+        .pull_down_en = GPIO_PULLDOWN_DISABLE,
+        .intr_type = GPIO_INTR_DISABLE,
```

Comment:
Here are you configuring the `XPT2046_IRQ` pin as input only and not as interrupt isn't?

I don't know if lvgl have limitations when reading input devices, maybe in the future we can try configuring the `XPT2046_IRQ` to trigger an interrupt on a touch event.

Reply:
This one is tricky. Although the pin is called an interrupt signal, it is asserted both when the screen is touched and then later when the value is read. This [post](https://github.com/PaulStoffregen/XPT2046_Touchscreen) gives a pretty good explanation of the situation 

> However, inside your interrupt function, if the display is no longer being touched, any attempt to read the touch position will cause the interrupt pin to create another falling edge. This can lead to an infinite loop of falsely triggered interrupts. Special care is needed to avoid triggering more interrupts on the low signal due to reading the touch position:

> For most applications, regularly reading the touch position from the main program is much simpler.

The library's current approach of polling the interrupt line to see if the screen is being touched and only invoking the SPI transaction if it is seems like a fairly good option, albeit not as CPU efficient as a purely interrupt driven solution.

Reply:
You're right, lvgl reads the input device periodically to know it's state.